app/src/main/java/at/aau/serg/activities/LobbyActivity.kt

```diff
@@ -0,0 +1,147 @@
+        SocketHandler.on("lobby:userLeft", ::userLeft)
+        SocketHandler.on("lobby:userKick", ::userKick)
+    }
+
+       private fun onSuccessGetLobby(response: Response) {
+        if (response.isSuccessful) {
+            val responseBody = response.body?.string()
+            if (responseBody != null) {
+                Log.d("Lobby", responseBody)
+                val players = Array(JSONObject(responseBody).getJSONArray("players").length()) {
+                    JSONObject(responseBody).getJSONArray("players").getString(it)
+                }
+                for ((index, player) in players.withIndex()) {
+                    Log.d("Lobby", player.toString())
+                    lobbyPlayers[index].uuid = JSONObject(player).getString("uuid")
+                    lobbyPlayers[index].name = JSONObject(player).getString("username")
+                    adapter.notifyItemChanged(index)
+                }
+                val adminUUID: String = JSONObject(players[0]).getString("uuid")
+                val uuid: String = StoreToken(this).getUUID().toString()
+                if(adminUUID == uuid){
+                    isAdmin = true
+                }
+
+                Log.d("Lobby", adminUUID)
+                Log.d("Lobby", uuid)
+            }
+        }
+    }
+
+    private fun onFailure() {
+        this.startActivity(Intent(this, MainActivity::class.java))
+    }
+
+    private fun userJoined(anies: Array<Any>) {
+        val playerData = (anies[0] as JSONObject)
+        Log.d("Lobby", "$playerData joined")
+
+        for ((index, player) in lobbyPlayers.withIndex()) {
+            if (player.uuid == "") {
+                player.uuid = playerData.getString("playerUUID")
+                player.name = playerData.getString("playerName")
+                if(isAdmin){
+                    player.isVisible = 0
+                }
+                adapter.notifyItemChanged(index)
+                break
+            }
+        }
+    }
+
+    private fun userLeft(anies: Array<Any>) {
+        val playerData = (anies[0] as JSONObject).getString("playerUUID")
+        Log.d("Lobby", "$playerData left")
+        val plyerToRemove = lobbyPlayers.indexOfFirst { it.uuid == playerData }
+        lobbyPlayers[plyerToRemove].uuid = ""
+        lobbyPlayers[plyerToRemove].name = ""
+        lobbyPlayers[plyerToRemove].isVisible =4
+        adapter.notifyItemChanged(plyerToRemove)
+    }
+
+    private fun userKick(anies: Array<Any>) {
+        startActivity(Intent(this, MainActivity::class.java))
+    }
+
+    fun btnLeaveClicked(view: View) {
+        HttpClient.get(
+            "/lobbys/leave",
+            StoreToken(this).getAccessToken(),
+            CallbackCreator().createCallback(::onFailureCLobby, ::leftLobby)
+        )
+    }
+
+    private fun leftLobby(response: Response) {
+        startActivity(Intent(this, MainActivity::class.java))
+    }
+
+    private fun onFailureCLobby() {
```

Comment:
Rename the function to something more descriptive and add some error handling or at least a Toast message